### PR TITLE
Sanitize mark_safe usages for reports

### PR DESCRIPTION
**Pull request by mjriley**
## Summary
Another XSS ticket as part of [SAAS-11853](https://dimagi-dev.atlassian.net/browse/SAAS-11853)

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [ ] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Small test suite in test_models.py

### QA Plan

This will need to be QA'd in conjunction with the data tables change to verify that the reports are still displaying correctly (no incorrectly-encoded HTML characters like ', ", &, <, and >)

### Safety story
Some areas were local tested where possible. The biggest concern is how this integrates with the data tables change.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/ex-submodules/phonelog/reports.py [hunk 1]
```diff
@@ -241,24 +241,25 @@ def _create_row(self, log, matching_id, _device_users_by_xform, user_query, devi
...
-            "tooltip": "Show the surrounding 100 logs."
-        }
+        log_tag_format = format_html(
+            self._log_tag_fmt,
+            url="%s?goto=%s" % (self.get_url(domain=self.domain),
+                html.escape(json.dumps(log.id))),
+            classes=" ".join(tag_classes),
+            text=log_tag,
+            extra_params=(' data-datatable-highlight-closest="tr"' if log.id == matching_id else ''),
+            tooltip="Show the surrounding 100 logs."
```
Comment (kaapstorm) [inline corehq/ex-submodules/phonelog/reports.py:251]: Maybe out of scope for this PR, but shouldn't this be translated?

File: corehq/apps/reports/filters/users.py [hunk 2]
```diff
@@ -26,6 +27,9 @@
     BaseSingleOptionFilter,
 )
 
+#TODO: replace with common code
+mark_safe_lazy = lazy(mark_safe, str)
```
Comment (kaapstorm) [inline corehq/apps/reports/filters/users.py:31]: Would `corehq/util/markup.py` be a good home for this? (Is that kinda what you mean by "replace with common code"?)

File: corehq/apps/reports/filters/forms.py [hunk 3]
```diff
@@ -38,6 +39,9 @@
 PARAM_VALUE_STATUS_ACTIVE = 'active'
 PARAM_VALUE_STATUS_DELETED = 'deleted'
 
+# TODO: Replace with library method
+mark_safe_lazy = lazy(mark_safe, str)
```
Comment (kaapstorm) [inline corehq/apps/reports/filters/forms.py:43]: (See other comment re `corehq/util/markup.py`.)

**Review by kaapstorm — APPROVED** [review]

File: corehq/ex-submodules/phonelog/reports.py (hunk 1 above)
Comment (mjriley) [reply corehq/ex-submodules/phonelog/reports.py:251]: Translation would make sense. But yes, the scope of this PR is essentially just finding all the places where data tables aren't sanitizing their output and updating them. I'd generally agree with going in and making small changes to leave the code better than I found it, but I think you can see, with the amount of changes required all over the place, that I want to keep the scope as narrow as possible here.

File: corehq/apps/reports/filters/users.py (hunk 2 above)
Comment (mjriley) [reply corehq/apps/reports/filters/users.py:31]: That would be a good candidate. I believe I had two branches going simultaneously, where one had created a separate module for this including tests. While this branch was also in flight, I left the comment to switch these over to use that module, when the other branch is merged. I'll create a reminder for myself to remember to do that, but with these changes due on the 12th, I likely won't have a chance to address them prior

**Pull request merged**